services/github/github-package-registry-downloads.service.js

```diff
@@ -0,0 +1,295 @@
+    messageSuffix: '/today',
+  },
+  week: {
+    transform: statistics => statistics.downloadsThisWeek,
+    messageSuffix: '/week',
+  },
+  month: {
+    transform: statistics => statistics.downloadsThisMonth,
+    messageSuffix: '/month',
+  },
+  year: {
+    transform: statistics => statistics.downloadsThisYear,
+    messageSuffix: '/year',
+  },
+  total: {
+    transform: statistics => statistics.downloadsTotalCount,
+    messageSuffix: '',
+  },
+}
+
+module.exports = class GithubPackageRegistryDownloads extends GithubAuthV4Service {
+  static get category() {
+    return 'downloads'
+  }
+
+  static get route() {
+    return {
+      base: 'github/packages/downloads',
+      pattern:
+        ':user/:repo/:interval(today|week|month|year|total)/:packageName/:version?',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub Package Registry Downloads',
+        pattern:
+          ':user/:repo/:interval(today|week|month|year|total)/:packageName',
+        namedParams: {
+          user: 'github',
+          repo: 'auto-complete-element',
+          interval: 'total',
+          packageName: 'auto-complete-element',
+        },
+        staticPreview: this.render({
+          interval: 'total',
+          count: 50,
+        }),
+        documentation,
+      },
+      {
+        title: 'GitHub Package Registry Downloads (Version)',
+        pattern:
+          ':user/:repo/:interval(today|week|month|year|total)/:packageName/:version',
+        namedParams: {
+          user: 'github',
+          repo: 'auto-complete-element',
+          interval: 'total',
+          packageName: 'auto-complete-element',
+          version: '1.0.6',
+        },
+        staticPreview: this.render({
+          interval: 'total',
+          count: 5,
+          version: '1.0.6',
+        }),
+        documentation,
+      },
+      {
+        title: 'GitHub Package Registry Downloads (Latest Version)',
+        pattern:
+          ':user/:repo/:interval(today|week|month|year|total)/:packageName/:version',
+        namedParams: {
+          user: 'github',
+          repo: 'auto-complete-element',
+          interval: 'total',
+          packageName: 'auto-complete-element',
+          version: 'latest',
+        },
+        staticPreview: this.render({
+          interval: 'total',
+          count: 9,
+          version: 'latest',
+        }),
+        documentation,
+      },
+    ]
+  }
+
+  static _getLabel(version) {
+    if (version) {
+      return `downloads@${version}`
+    } else {
+      return 'downloads'
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'downloads' }
+  }
+
+  static render({ interval, count, version }) {
+    const { messageSuffix } = intervalMap[interval]
+    return {
+      label: this._getLabel(version),
+      message: `${metric(count)}${messageSuffix}`,
+      color: downloadCount(count),
+    }
+  }
+
+  async fetch({ user, repo, packageName, version }) {
```

Comment:
This function got a little long, but that's mostly because of the formatting and the fact that we need three different GraphQL queries. 

I suspect there's away that the queries/variables could be collapsed into one that is conditionally constructed based on the params (for example including the `version` filter), but I suppose having them inline like this helps with readability.